### DISCUSSION TITLE
RustDesk creates world-read/write directory, socket and PID file in /tmp

### DISCUSSION THREAD

**Discussion by wodin** [rustdesk/rustdesk#4324] [General]
**Describe the bug you encountered:**

RustDesk creates a world-read/write `/tmp/RustDesk` directory containing a world-writable `ipc` socket and a world-writable `ipc.pid` file.

```
# ls -ld /tmp/RustDesk /tmp/RustDesk/*
drwxrwxrwx 2 root root 4 Oct 21 12:43 /tmp/RustDesk
srwxrwxrwx 1 root root 0 Oct 21 12:43 /tmp/RustDesk/ipc
-rwxrwxrwx 1 root root 6 Oct 21 12:43 /tmp/RustDesk/ipc.pid
```

**What did you expect to happen instead?**

I expect services to use `/run/*` for keeping track of PIDs and sockets.
Also, the PID file should not be writable by all users on the system and it might not need to be readable either.
Also, I am not sure if the socket needs to be readable/writable by all users on the system.
Of less importance: I don't think there is any reason for the PID file and the socket to have the execute bits set.

**How did you install `RustDesk`?**

I installed the deb that I downloaded from here:
http://github.com/rustdesk/rustdesk/releases/download/1.1.9/rustdesk-1.1.9.deb

---

**RustDesk version and environment**

```
# dpkg -l rustdesk 
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version      Architecture Description
+++-==============-============-============-=================================
ii  rustdesk       1.1.9        amd64        A remote control software.
```

```
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.1 LTS
Release:	22.04
Codename:	jammy
```

**Comment by foodornt** [comment]
I hope rustdesk team takes a look into this issue, `/tmp` folder is being polluted

**Comment by nobody43** [comment]
As far as I understand it's needed for userland GUI and root-based service intercommunication. And this makes it not trivial to solve. Either way, this is not an excuse not to fix this serious security issue.

**Comment by wodin** [comment]
I assume you're talking about the socket. Perhaps the GUI could be setgid `rustdesk` and the socket could be writable only by that group?

**Reply by wodin** [reply]
At minimum these things should be moved to `/run` and remove the executable bits from the socket and PID file.

Could someone explain why the `RustDesk` directory needs to be world writable? Also the PID file?
Also, what exactly needs to be able to write to the socket?

**Comment by SergeyMy** [comment]
https://github.com/rustdesk/rustdesk/issues/4327
Is the problem related to this as well?

**Comment by servel333** [comment]
Check your versions. I seem to have gotten this because my two versions were different.

For the Rustdesk team, please add a specific message when there is a version mismatch between the two clients.
